src/modules/inventory/components/AddProductForm.tsx

```diff
@@ -37,22 +42,29 @@ const AddProductForm: React.FC<Props> = ({ product }) => {
         images: yup.array().of(yup.string()).required("must be at least 1").min(1, "must be at least 1").max(5, "must be at most 5"),
     });
 
-    const { register, handleSubmit, setValue, formState: { errors }, reset, watch } = useForm<IProduct>({
-        resolver: yupResolver(productSchema)
+    const { register, handleSubmit, reset, setValue, watch, formState: { errors } } = useForm<IProduct>({
+        resolver: yupResolver(productSchema),
+        defaultValues: product || {}
     });
 
-    const [loading, setLoading] = useState<boolean>(false);
-    const [components, setComponents] = useState<IComponent[]>([]);
-    const [imagePreviews, setImagePreviews] = useState<string[]>([]);
-    const dispatch = useDispatch();
-    const componentState = useSelector((state: RootState) => state.component);
-
     useEffect(() => {
-        if (product) {
-            reset(product);
-        }
-    }, [product, reset]);
-
+        const fetchProduct = async () => {
+            if (productId) {
+                try {
+                    const fetchedProduct = await getItemById<any>(`api/inventory/product`, productId);
+                    const upDateProduct = fetchedProduct;
+                    delete upDateProduct.data._id;
+                    delete upDateProduct.data.__v;
+                    setProduct(upDateProduct);
```

Comment:
עדיף ליצור משתנה חדש בלי השדות האלו
לא מקובל למחוק נתונים שקיבלנו מהbackend

Reply:
את מתכוונת שאני אשנה בקריאת שרת שתחזיר לי את האובייקט ללא השדות הללו?
כלומר בפרויקט של Orders?

Reply:
לא, ככה
const dataToUpdate = {...updateProduct, _id: undefined, __v: undefined})

Reply:
שינתי את זה ועדין עושה לי בעיה כי א"א לשלוח את השדות v__ , _id לשרת בעדכון כי הם אמורים להתווסף אוטומטית
זה עושה לי את השגיאה הזו לאחר השינוי הנ"ל
[Nest] 29  - 07/30/2024, 10:30:22 AM   ERROR [ExceptionsHandler] "_id" is not allowed
2024-07-30 13:30:22 ValidationError: "_id" is not allowed
מה אפשר לעשות?

Reply:
ניסית את הקוד שנתתי לך? וזה מחזיר לך שגיאה? תראי לי איך השתמשת בו

Reply:
כן עשה לי שגיאה בצורה הזו
שיניתי לדרך הזה
                    const fetchedComponent = await getItemById<any>(`api/inventory/component`, componentId);
                    **const { _id, __v, ...dataToUpdate } = fetchedComponent.data;**
                    setComponent(dataToUpdate);
                    reset(dataToUpdate);

ועכשיו זה עובד ללא שגיאות, הדרך הזאתי בסדר זה עובר?

Reply:
ההצעה שלך מצוינת